conversion/conversion.go

```diff
@@ -284,54 +284,95 @@ func processDataWithDataproc(sourceProfile profiles.SourceProfile, targetProfile
 	conv.Audit.DataprocMetadata.DataprocJobUrls = make(map[string]string)
 	conv.Audit.DataprocMetadata.DataprocJobIds = make(map[string]string)
 	conv.Audit.DataprocMetadata.DataprocJobStatus = make(map[string]string)
-	for _, spannerTableID := range orderTableNamesByID {
-		srcTable := conv.SrcSchema[spannerTableID].Name
 
-		srcSchema := conv.SrcSchema[spannerTableID]
+	var processTable func(spannerTableID string) error
+	processTable = func(spannerTableID string) error {
+		parentTableId := conv.SpSchema[spannerTableID].ParentId
 
-		primaryKeys, _, _ := infoSchema.GetConstraints(conv, common.SchemaAndName{Name: srcTable, Schema: srcSchema.Schema})
-
-		dataprocRequestParams, err := dproc.GetDataprocRequestParams(sourceProfile, targetProfile, srcSchema.Schema, srcTable, strings.Join(primaryKeys, ","), location, subnet)
-		if err != nil {
-			return err
+		// TODO: Replace DataprocJobStatus values to ENUMS / CONSTANTS
+		wait := true
+		if parentTableId == "" {
+			wait = false
+		} else {
+			parentTableStatus := conv.Audit.DataprocMetadata.DataprocJobStatus[parentTableId]
+			if parentTableStatus == "SUCCESS" {
+				wait = false
+			} else if parentTableStatus == "FAILED" || parentTableStatus == "SKIPPED" {
```

Comment:
Lets define constants and use them

Reply:
Done,
Commit: https://github.com/GoogleCloudPlatform/spanner-migration-tool/pull/611/commits/bd106749bfe5174e9606c83ce5f3b45a9bdda67d